### ISSUE TITLE
Pressing space after a single character too quickly causes a period to be entered

### ISSUE THREAD

**Issue by drcouzelis** [florisboard/florisboard#170]
Using Floris Board 0.3.2, the lastest beta version from GitHub at the time of this post.

I thought I was going crazy because random periods kept being typed! Here is my guess as to what is happening with an example sentence...

Typing "I ate a pizza" becomes "I ate . Pizza"

A period is automatically entered when pressing space twice in a short amount of time, even if another character is entered inbetween the spaces.

In other words, pressing "space-a-space" produces a period.

Does that make sense? It happens to me all the time because typing with Floris Board is so fast!

Thank you. :-)

**Comment by GithbUser** [comment]
Yes you are right 
I confirm this bug 
But this happens only when you type very fast

**Comment by RickyM7** [comment]
This is a feature of FlorisBoard, this behavior is expected but you can disable this.

![period](https://user-images.githubusercontent.com/24703825/104282572-04599100-548e-11eb-991c-fcd65ca8a348.jpg)

**Comment by patrickgold** [comment]
@drcouzelis Yes, it is intended that pressing space two times in a row (with a short delay in between) inserts a space followed by a period (`.`). This behavior can be toggled in the Typing tab of the Settings (screenshot above).

What is definitely a bug though is that pressing `space` `a` `space` very quickly results in `<space>.<space>`, rather than the expected `<space>a<space>`. I will put this on my todo list to fix!

**Comment by drcouzelis** [comment]
Thank you! :) FYI I found another issue that I think is similar enough to mention here...

Action: Begin a new sentence by typing "O, shift, M, shift, G".

Expected result: "OMG"

Actual result: "OmG", and the shift key is now in sticky mode.

I assume, once again, the double-press of the shift key is ignoring the characters that are being pressed inbetween.

...your excellent keyboard has made me realize I type "OMG" a lot. XD

**Comment by patrickgold** [comment]
I see my double-press detection timers are not fail-safe for fast writers :)

> I assume, once again, the double-press of the shift key is ignoring the characters that are being pressed inbetween.

I also think that it has the same reason as the space punctuation issue as it uses the same method of detecting a double press. I will investigate into this further.

**Comment by patrickgold** [comment]
@drcouzelis Above commit fixes the double-press bugs for space and shift by resetting the double key detection state when a character is entered or if the delete/enter key is pressed. Will be shipped with v0.3.4 either tomorrow or Thursday evening. Thanks again for discovering and reporting this bug!
